templates/workflows/update-wp-version.yml

```diff
@@ -0,0 +1,122 @@
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.ref }}
+
+      - name: Set Variables
+        id: vars
+        run: |
+          echo "sha_short=$(git rev-parse --short ${{ github.sha }})" >> $GITHUB_OUTPUT
+          echo "timestamp=$(date +%Y%m%d-%H%M%S)" >> $GITHUB_OUTPUT
+
+      - name: Set up Git configuration
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "github-actions"
+
+      - name: Create branch for changes
+        id: create-branch
+        run: |
+          BRANCH_NAME="task/update-wp-version-${{ steps.vars.outputs.timestamp }}-${{ steps.vars.outputs.sha_short }}"
+          git checkout -b "$BRANCH_NAME"
+          git push --set-upstream origin "$BRANCH_NAME"
+          echo "branch_name=$BRANCH_NAME" >> $GITHUB_OUTPUT
+
+      - name: Calculate and update versions
+        id: update-versions
+        run: |
+          TESTED_VERSION="${{ github.event.inputs.tested_up_to }}"
+          IFS='.' read -r -a tested_parts <<< "$TESTED_VERSION"
+
+          # Calculate minimum version (2 minor versions behind, using only major.minor)
+          min_major=$((tested_parts[0]))
+          min_minor=$((tested_parts[1] - 2))
+          MIN_VERSION="$min_major.$min_minor"
+
+          echo "## Version Updates" >> $GITHUB_STEP_SUMMARY
+          echo "* Tested Up To: \`$TESTED_VERSION\`" >> $GITHUB_STEP_SUMMARY
+
+          if [ "${{ github.event.inputs.update_min_version }}" = "yes" ]; then
+            echo "* Minimum Required Version: \`$MIN_VERSION\`" >> $GITHUB_STEP_SUMMARY
+          else
+            echo "* Minimum Required Version: Unchanged" >> $GITHUB_STEP_SUMMARY
+          fi
+
+          # Update readme.txt
+          if [ -f "readme.txt" ]; then
+            sed -i "s/Tested up to: [0-9.]*/Tested up to: $TESTED_VERSION/" readme.txt
+            if [ "${{ github.event.inputs.update_min_version }}" = "yes" ]; then
+              sed -i "s/Requires at least: [0-9.]*/Requires at least: $MIN_VERSION/" readme.txt
+            fi
+            echo "* Updated \`readme.txt\`" >> $GITHUB_STEP_SUMMARY
+          fi
+
+          # Update the-events-calendar.php if minimum version is being updated
+          if [ "${{ github.event.inputs.update_min_version }}" = "yes" ] && [ -f "the-events-calendar.php" ]; then
+            sed -i "s/Requires at least: [0-9.]*/Requires at least: $MIN_VERSION/" the-events-calendar.php
+            echo "* Updated \`the-events-calendar.php\`" >> $GITHUB_STEP_SUMMARY
+          fi
+
+          # Update tests-php.yml and other workflow files if minimum version is being updated (skips self)
+          if [ "${{ github.event.inputs.update_min_version }}" = "yes" ] && [ -d ".github/workflows" ]; then
+            for wf in .github/workflows/*.yml; do
+              [ "$(basename "$wf")" = "update-wp-version.yml" ] && continue
+              sed -i "s/--version=[0-9.]*/--version=$MIN_VERSION/" "$wf"
```

Comment:
maybe a bit more specific regex here would help ? `wp core update` it's always part of the command we want to target

Reply:
you could actually just target the whole lines that matches `^wp core update (.*)$` and simply replace it with any format you desire => `wp core update --version=$MIN_VERSION --force`

Reply:
That's a good idea - this was the only use of `version=` I found, but that doesn't mean another won't come along.